dev/hm10/export/dev/hm10.hpp

```diff
@@ -0,0 +1,199 @@
+//! \file
+//! \brief HM-10 BT module drivers.
+//! Header must expose both synchronous (simple but not so functional) and
+//! asynchronious versions of the driver.
+//! At this moment, only synchronous version is implemented.
+
+#ifndef DEV_BT_HM10_HPP_
+#define DEV_BT_HM10_HPP_
+
+#include <ecl/err.hpp>
+#include <ecl/assert.h>
+
+#include <stdint.h>
+#include <stddef.h>
+#include <string.h>
+
+#include <algorithm>
+#include <chrono>
+
+namespace ecl
+{
+
+//! HM10 synchronous BT driver.
+//! \details This HM10 driver is simple and low-functional API to send and receive
+//! data from BT module. It lacks connection status detection and do not escape
+//! potential AT commands found in payload (yet). No flow control is used
+//! whatsoever.
+//! External protocol is required to control whether other end is ready to
+//! receive or send a data. Great advantage of this particular driver is its API.
+//! Straitforward and blocking it can be used for simple demo implementations.
+//! \note Driver _is_not_ a thread safe. Using driver routine from ISR context
+//! will lead to undefined behaviour.
+//! \tparam UART bus to work with.
+template<typename Uart>
+class hm10_sync
+{
+public:
+    //! Initializes BT module.
+    //! \return Status of operation.
+    //! \retval err::inval      Module respond with invalid data.
+    //! \retval err::timedout   Module did not respond at all.
+    //! \retval err::ok         Module is ready to work.
+    static err init();
+
+    //! Sends data to the module.
+    //! \warning It is possible that no one has connected to the device yet.
+    //! \warning Data that is sent to unconnected module will be lost.
+    //! \details Call blocks until all data will be sent.
+    //! \param[in] buf Buffer to send.
+    //! \param[in] sz  Size of buffer.
+    //! \return Status of operation.
+    //! \retval err::ok         Module is ready to work.
+    static err send(const uint8_t *buf, size_t sz);
+
+    //! Receives data from the module with given timeout.
+    //! \param[out]     buf Buffer to store incoming data.
+    //! \param[in,out]  sz  Size of the buffer as input, bytes effectively written
+    //!                     as output.
+    //! \param[in]      ms  Time to wait for incoming data.
```

Comment:
I propose to document that the function will block infinitely if the default value for `ms` argument is used.